### PR TITLE
feat: handle cwd and exe as symlinks

### PR DESCRIPTION
**Pull request by irozzo-1A**

### CODE REVIEW

File: src/async.cpp [hunk 1]
```diff
@@ -138,8 +142,27 @@ static int fuse_read(const char *path,
...
+	std::unique_lock l(ctx->m_mu);
+	ctx->done = false;
+	memset(buf, 0, size);
+	ctx->buf = buf;
+	int pid = 0;
+	char entry[32];
+	sscanf(path, "/%d/%s", &pid, entry);
+	generate_async_event(ctx->async_event_handler, ASYNC_EVENT_ENTRY_NAME, pid, entry);
+	ctx->m_cv.wait(l, [ctx] { return ctx->done; });
+	return 0;
```
Comment (Andreagit97) [inline src/async.cpp:160]: why do we return `0` and not `strlen(buf) + 1;` like in `fuse_read`?

Comment (irozzo-1A) [reply src/async.cpp:160]: @Andreagit97 the return value is 0 for success https://github.com/libfuse/libfuse/blob/7a92727d97c10290b3501d86a194738973edb61d/include/fuse.h#L338-L346

Comment (irozzo-1A) [reply src/async.cpp:160]: error handling could be improved though 😄

Comment (Andreagit97) [reply src/async.cpp:160]: ok thank you

**Pull request merged**